app/assets/javascripts/project_find_file.js.coffee

```diff
@@ -0,0 +1,110 @@
+class @ProjectFindFile
+  constructor: (@options)->
+    @filePaths = {}
+    @element = ".file-finder-input"
+    @row = "<tr class='tree-item'><td class='tree-item-file-name'><i class='fa fa-file-text-o fa-fw'></i><span class='str-truncated'><a href=%blobItemUrl>%filePath</a></span></td></tr>"
+
+    # init key event(shortcut, find file)
+    @initKeyEvent()
+
+    # focus text input box
+    $(@element).focus()
+
+    # load file list
+    @load(@options.url)
+
+  # files pathes load
+  load: (url) ->
+    $.ajax
+      url: url
+      method: "get"
+      dataType: "json"
+      success: $.proxy((data) ->
+        $(".loading").hide()
+        @filePaths = data
+        @findFile($(@element).val())
+        @selecteRow $("tr.tree-item")
+      , this)
+
+  # find file
+  findFile: (searchTxt) ->
+      result = if searchTxt then fuzzaldrinPlus.filter(@filePaths, searchTxt) else @filePaths
+      @renderList result, searchTxt
+
+  # render result
+  renderList: (filePaths, searchTxt) ->
+    $(".tree-table > tbody").empty()
+    for filePath, i in filePaths
+      break if i > 20
+      markedFilePath = if searchTxt then highlightText filePath, searchTxt.split("") else filePath
+      blobItemUrl = "#{@options.blobUrlTemplate}/#{filePath}"
+      html = @makeHtml markedFilePath, blobItemUrl
+      $(".tree-table > tbody").append(html)
+
+  # highlight text(awefwbwgtc -> <b>a</b>wefw<b>b</b>wgt<b>c</b> )
+  highlightText = (txt, charArr) ->
+    result = ""
+    target = txt.toString()
+    for char in charArr
+      charIdx = target.toLowerCase().indexOf char.toLowerCase()
+      result += "#{target.substring 0, charIdx}#{(target.charAt charIdx).bold()}"
+      target = target.substring charIdx + 1
+    return (result + target)
+
+  # make tbody row html
+  makeHtml: (filePath, blobItemUrl) ->
+    return  @row.replace("%blobItemUrl", blobItemUrl).replace("%filePath", filePath)

```

Comment:
What happens if the file path has the text `%blobItemUrl`? Can we just construct the entire text without a `replace`?